moz/l10n/resource/data.py

```diff
@@ -38,6 +38,34 @@
 """
 
 
+@dataclass
+class LinePos:
+    """
+    The source line position of an entry or section header.
+    """
+
+    start: int
+    """
+    The starting line of the entry or section.
+    May be less than `value` if preceded by a comment.
+    """
+
+    key: int
+    """
+    The start line of the entry or section header key or name.
+    """
+
+    value: int
+    """
+    The start line of the entry pattern or section header.
+    """
+
+    end: int
+    """
+    The line one past the end of the entry or section header.
```

Comment:
Why "one past"? What happens at the end of a file if there's no empty line?

Reply:
Then this'll point to a line that's not there. It's the same as with character ranges; the "end" pointer is not included in the range. This works even if there's no such character (or line) actually there.